src/smf/s5c-handler.c

```diff
@@ -78,8 +78,8 @@ uint8_t smf_s5c_handle_create_session_request(
         ogs_error("No IMSI");
         cause_value = OGS_GTP2_CAUSE_CONDITIONAL_IE_MISSING;
     }
-    if (req->msisdn.presence == 0) {
-        ogs_error("No MSISDN");
+    if (req->imsi.presence == 0 && req->msisdn.presence == 0) {
```

Comment:
Fine with having this in GTPv2C, tbh the specs is quite vague here on why the MSISDN may or may not come. It seems you inferred  somehow that if the IMSI is provided it means that the HSS was fetched.

Reply:
As per 3GPP TS 29.274 version 16.5.0, Create Session Request, (following snippet corresponding to MSISDN)

![image](https://user-images.githubusercontent.com/4115839/167446918-225e63ea-12ff-45ca-b3e1-0caa8c0212ba.png)

Spec is clear that IMSI would be present for sure in Create Session Request, however in the snippet above you see that presence of MSISDN presence is based on the presence of MSISDN in ULA received from HSS towards MME.

Reply:
Yes, I can see the spec for GTV2C seems to describe that CreateSessionReq may not come with MSISDN, but I don't see how the fact that the IMSI being present or not has anything to do with expecting the MSISDN being present or not.
In here ("_if (req->imsi.presence == 0 && req->msisdn.presence == 0) {_"), however, you seem to describe that they are somehow related. That's the part I'm not following  (I'm not saying it's wrong, it just looks strange without proper explanation).

Reply:
ACK fine with me.